### PR TITLE
I102 protect user and team dashboard pages

### PR DESCRIPTION
**Pull request by Alex-Hawking**
## Change Summary
**[Briefly summarise the changes that you made. Just high-level stuff]**
Made it so that use cannot redirect to team and dashboard pages if not logged
Login modal opens if redirect fails

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [x] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

File: client/src/components/LoginModal.vue [hunk 1]
```diff
@@ -196,9 +196,18 @@ import { ref, watchEffect } from 'vue'
 import { useDisplay } from 'vuetify'
 import { useUserStore } from '@/stores/user'
 import { AxiosError } from 'axios'
-
+import { useModalStateStore } from '@/stores/openModal'
 const { mobile } = useDisplay()
```
Comment (ChantelleKerr) [inline client/src/components/LoginModal.vue:200]: we can remove the line with useDisplay

File: client/src/components/LoginModal.vue [hunk 2]
```diff
@@ -196,9 +196,18 @@ import { ref, watchEffect } from 'vue'
...
+
+import { useRouter } from 'vue-router'
+const router = useRouter()
```
Comment (ChantelleKerr) [inline client/src/components/LoginModal.vue:203]: Since this is not used we can remove it

File: client/src/components/LoginModal.vue [hunk 3]
```diff
@@ -196,9 +196,18 @@ import { ref, watchEffect } from 'vue'
...
+
+const ModalStateStore = useModalStateStore()
```
Comment (ChantelleKerr) [inline client/src/components/LoginModal.vue:205]: is this used anywhere?

File: client/src/components/LoginModal.vue [hunk 4]
```diff
@@ -196,9 +196,18 @@ import { ref, watchEffect } from 'vue'
...
+
 const dialog = ref(true)
+
+console.log(dialog.value)
+
```
Comment (ChantelleKerr) [inline client/src/components/LoginModal.vue:210]: remove console log

**Review by ChantelleKerr — CHANGES_REQUESTED** [review]
very quick fixes and I can approve.

Good job!

**Comment by Alex-Hawking** [comment]
All changes should be done

**Review by ChantelleKerr — APPROVED** [review]
Good job! Approved

**Pull request merged**